### PR TITLE
Load plugin schemas after built-in schemas

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This change allows for plugins to override bundled schemas as part of their installation. The plugin schemas reside in `<datadir>/vast/plugin/<plugin>/schema`, while the builtin schemas reside in `<datadir>/vast/schema`. The load order guarantees that plugins are able to reliably override the schemas bundled with VAST.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Commit-by-commit.

### CODE REVIEW

File: vast/vast.cpp [hunk 1]
```diff
@@ -57,11 +57,16 @@ int main(int argc, char** argv) {
...
-  for (auto&& plugin_names : std::vector<std::string>{VAST_ENABLED_PLUGINS})
-    if (std::none_of(plugin_paths_or_names.begin(), plugin_paths_or_names.end(),
-                     [&](auto&& x) { return x == plugin_names; }))
-      plugin_paths_or_names.push_back(std::move(plugin_names));
+  // if they were not already configured as part of vast.plugins, unless
+  // --disable-default-config-dirs is set, which would break the integration
+  // tests that rely on this feature to avoid accidentally loading undesired
+  // plugins.
+  if (!caf::get_or(cfg, "vast.disable-default-config-dirs", false))
+    for (auto&& plugin_names : std::vector<std::string>{VAST_ENABLED_PLUGINS})
```
Comment (lava) [inline vast/vast.cpp:65]: s/plugin_names/plugin_name/

File: libvast/src/schema.cpp [hunk 2]
```diff
@@ -219,23 +220,34 @@ caf::expected<schema> get_schema(const caf::settings& options) {
...
+      std::error_code err{};
+      auto dir = datadir / "plugin" / plugin->name() / "schema";
+      if (auto err = std::error_code{}; std::filesystem::exists(dir, err))
+        result.insert(std::move(dir));
+    }
+  };
 #if !VAST_ENABLE_RELOCATABLE_INSTALLATIONS
-  result.insert(VAST_SCHEMADIR);
+  insert_dirs_for_datadir(VAST_DATADIR);
 #endif
```
Comment (lava) [inline libvast/src/schema.cpp:241]: Not the scope of this PR, but shouldn't the loop below be in an `#else` block?

File: libvast/src/schema.cpp [hunk 3]
```diff
@@ -219,23 +220,34 @@ caf::expected<schema> get_schema(const caf::settings& options) {
...
   detail::stable_set<std::filesystem::path> result;
   if (auto vast_schema_directories = detail::locked_getenv("VAST_SCHEMA_DIRS"))
     for (auto&& path : detail::split(*vast_schema_directories, ":"))
       result.insert({path});
+  auto insert_dirs_for_datadir = [&](const std::filesystem::path& datadir) {
+    result.insert(datadir / "schema");
+    for (const auto& plugin : plugins::get()) {
+      std::error_code err{};
+      auto dir = datadir / "plugin" / plugin->name() / "schema";
+      if (auto err = std::error_code{}; std::filesystem::exists(dir, err))
```
Comment (lava) [inline libvast/src/schema.cpp:235]: You define `err` twice. (personally the variant where it's defined outside the if-statement looks a bit easier to read)

**Review by lava — APPROVED** [review]
Code looks good apart from some minor comments.

I didn't personally test this, but I trust you already did while working on the matcher.

File: libvast/src/schema.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/src/schema.cpp:241]: That'd break integration tests from within the build directory.

**Pull request merged**